src/components/tx-flow/flows/EnableRecovery/EnableRecoveryFlowReview.tsx

```diff
@@ -0,0 +1,113 @@
+import { useContext, useEffect, useMemo } from 'react'
+import type { ReactElement } from 'react'
+
+import SignOrExecuteForm from '@/components/tx/SignOrExecuteForm'
+import { Errors, logError } from '@/services/exceptions'
+import { createMultiSendCallOnlyTx } from '@/services/tx/tx-sender'
+import { SafeTxContext } from '@/components/tx-flow/SafeTxProvider'
+import { getRecoverySetup } from '@/services/recovery/setup'
+import { useWeb3 } from '@/hooks/wallets/web3'
+import useSafeInfo from '@/hooks/useSafeInfo'
+import { SvgIcon, Tooltip, Typography } from '@mui/material'
+import { EnableRecoveryFlowFields, RecoveryDelayPeriods, RecoveryExpirationPeriods } from '.'
+import { TxDataRow } from '@/components/transactions/TxDetails/Summary/TxDataRow'
+import InfoIcon from '@/public/images/notifications/info.svg'
+import EthHashInfo from '@/components/common/EthHashInfo'
+import type { EnableRecoveryFlowProps } from '.'
+
+export function EnableRecoveryFlowReview({ params }: { params: EnableRecoveryFlowProps }): ReactElement {
+  const web3 = useWeb3()
+  const { safe } = useSafeInfo()
+  const { setSafeTx, safeTxError, setSafeTxError } = useContext(SafeTxContext)
+
+  const recoverySetup = useMemo(() => {
+    if (!web3) {
+      return
+    }
+
+    return getRecoverySetup({
+      ...params,
+      safe,
+      provider: web3,
+    })
+  }, [params, safe, web3])
+
+  useEffect(() => {
+    if (recoverySetup) {
+      createMultiSendCallOnlyTx(recoverySetup.transactions).then(setSafeTx).catch(setSafeTxError)
+    }
+  }, [recoverySetup, setSafeTx, setSafeTxError])
+
+  useEffect(() => {
+    if (safeTxError) {
+      logError(Errors._809, safeTxError.message)
+    }
+  }, [safeTxError])
+
+  const guardian = params[EnableRecoveryFlowFields.guardians][0]
+  const delay = RecoveryDelayPeriods.find(({ value }) => value === params[EnableRecoveryFlowFields.txCooldown])!.label
+  const expiration = RecoveryExpirationPeriods.find(
+    ({ value }) => value === params[EnableRecoveryFlowFields.txExpiration],
+  )!.label
+  const emailAddress = params[EnableRecoveryFlowFields.emailAddress]
+
+  return (
+    <SignOrExecuteForm onSubmit={() => null}>
```

Comment:
For some reason the decoded tx reloads every few seconds which retriggers the redefine scan.

Reply:
I cannot repliacte this. I have a feeling https://github.com/safe-global/safe-wallet-web/pull/2775#discussion_r1398244101 fixed it.

Reply:
It seems to be an issue with when we call `getRecoverySetup` since it depends on `safe` and `safe` updates every few seconds because one of the tags change. In my case its the `messagesTag`. Could this be related to the issue where the tags are updated if the queue/history is empty?

Reply:
I've adjusted `getRecoverySetup` to only accept the required values, as well as the dependencies in 7244e2e4432f56d3ce0355d9e2ebdfe383a652a5.